wallet/trezor-common/src/tests/mod.rs

```diff
@@ -0,0 +1,557 @@
+// Copyright (c) 2024 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{boxed::Box, println, vec::Vec};
+
+use common::{chain, primitives};
+use crypto::key::KeyKind;
+use parity_scale_codec::{DecodeAll, Encode};
+use rstest::rstest;
+use test_utils::random::{make_seedable_rng, CryptoRng, Rng, Seed};
+
+impl From<primitives::H256> for crate::H256 {
+    fn from(value: primitives::H256) -> Self {
+        Self(value.0)
+    }
+}
+
+impl From<chain::OutPointSourceId> for crate::OutPointSourceId {
+    fn from(value: chain::OutPointSourceId) -> Self {
+        match value {
+            chain::OutPointSourceId::Transaction(tx) => Self::Transaction(tx.to_hash().into()),
+            chain::OutPointSourceId::BlockReward(tx) => Self::BlockReward(tx.to_hash().into()),
+        }
+    }
+}
+
+impl From<chain::UtxoOutPoint> for crate::UtxoOutPoint {
+    fn from(value: chain::UtxoOutPoint) -> Self {
```

Comment:
Though I don't insist on this, I'd suggest to make the `make_random...` functions return a pair of values that are supposed to be encoded to the same bytes, and remove these `From` implementations (which look somewhat fragile to me).

Reply:
I think it is the same, because you would still need to put the same random data in the right fields for both versions when generating them.